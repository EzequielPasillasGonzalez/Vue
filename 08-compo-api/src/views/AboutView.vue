<template>
  <div class="about">
    <h1>This is an about page</h1>

    <h2>{{ title }}</h2>
  </div>

  <div>
    <button @click="changeTitle">Cambiar titulo</button>
  </div>
</template>


<script>
import { onActivated, onBeforeMount, onBeforeUnmount, onBeforeUpdate, onDeactivated, onErrorCaptured, onMounted, onRenderTracked, onUnmounted, onUpdated, ref } from 'vue';


export default {
  setup() {

    const title = ref('About page')

    console.log('setup se llamo');

    onBeforeMount(() => {
      console.log('beforeMount');
    })
      

    onMounted(() => {
      console.log('onMounted');
    })

    onBeforeUpdate(() => {
      console.log('onBeforeUpdate');
    })

    onUpdated(() => {
      console.log('onUpdated');
    })

    onBeforeUnmount(() => {
      console.log('onBeforeUnmount');
    })

    onUnmounted(() => {
      console.log('onUnmounted');
    })

    onErrorCaptured(() => {
      console.log('onErrorCaptured');
    })

    onRenderTracked(() => {
      console.log('onRenderTracked');
    })

    onRenderTracked(() => {
      console.log('onRenderTracked');
    })

    onActivated(() => {
      console.log('onActivated');
    })

    onDeactivated(() => {
      console.log('onDeactivated');
    })

    const changeTitle = () => {
      title.value = 'Titulo cambiado'
    }

    return {
      title,

      changeTitle
    }

  }
}
</script>

<style lang='scss' scoped>

</style>