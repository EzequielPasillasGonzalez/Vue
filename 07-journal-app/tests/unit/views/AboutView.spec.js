import { shallowMount } from '@vue/test-utils'

import AboutView from "@/views/AboutView.vue";

describe('Pruebas de en el aboutView', () => {

    let wrapper
    
    beforeEach(() => {
        wrapper = shallowMount(AboutView)
    })
    
    test('Debe de renderizar el componente correctamente', () => { 
        expect(wrapper.html()).toMatchSnapshot()
    })


    

})