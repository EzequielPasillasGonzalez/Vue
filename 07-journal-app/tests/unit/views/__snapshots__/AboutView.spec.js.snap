// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de en el aboutView Debe de renderizar el componente correctamente 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;

exports[`Pruebas de snapshot en el aboutView Debe de renderizar el componente correctamente 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;
